Description: when scraping magnet links, don't send a left=0 argument to the tracker.
Origin: https://trac.transmissionbt.com/changeset/13310
Bug: https://trac.transmissionbt.com/ticket/4888
Applied-Upstream: 2.60

Index: trunk/libtransmission/announcer.c
===================================================================
--- trunk/libtransmission/announcer.c	(revision 13300)
+++ trunk/libtransmission/announcer.c	(revision 13310)
@@ -913,5 +913,7 @@
     req->down = tier->byteCounts[TR_ANN_DOWN];
     req->corrupt = tier->byteCounts[TR_ANN_CORRUPT];
-    req->left = tr_cpLeftUntilComplete( &tor->completion );
+    req->leftUntilComplete = tr_torrentHasMetadata( tor )
+            ? tor->info.totalSize - tr_cpHaveTotal( &tor->completion )
+            : ~(uint64_t)0;
     req->event = event;
     req->numwant = event == TR_ANNOUNCE_EVENT_STOPPED ? 0 : NUMWANT;
Index: trunk/libtransmission/announcer-udp.c
===================================================================
--- trunk/libtransmission/announcer-udp.c	(revision 12982)
+++ trunk/libtransmission/announcer-udp.c	(revision 13310)
@@ -324,5 +324,5 @@
     evbuffer_add        ( buf, in->peer_id, PEER_ID_LEN );
     evbuffer_add_hton_64( buf, in->down );
-    evbuffer_add_hton_64( buf, in->left );
+    evbuffer_add_hton_64( buf, in->leftUntilComplete );
     evbuffer_add_hton_64( buf, in->up );
     evbuffer_add_hton_32( buf, get_tau_announce_event( in->event ) );
Index: trunk/libtransmission/announcer-common.h
===================================================================
--- trunk/libtransmission/announcer-common.h	(revision 12238)
+++ trunk/libtransmission/announcer-common.h	(revision 13310)
@@ -149,5 +149,5 @@
 
     /* the total size of the torrent minus the number of bytes completed */
-    uint64_t left;
+    uint64_t leftUntilComplete;
 
     /* the tracker's announce URL */
Index: trunk/libtransmission/announcer-http.c
===================================================================
--- trunk/libtransmission/announcer-http.c	(revision 13300)
+++ trunk/libtransmission/announcer-http.c	(revision 13310)
@@ -82,5 +82,5 @@
                               req->up,
                               req->down,
-                              req->left,
+                              req->leftUntilComplete,
                               req->numwant,
                               req->key );
Index: trunk/libtransmission/completion.h
===================================================================
--- trunk/libtransmission/completion.h	(revision 12918)
+++ trunk/libtransmission/completion.h	(revision 13310)
@@ -90,10 +90,4 @@
 }
 
-static inline uint64_t
-tr_cpLeftUntilComplete( const tr_completion * cp )
-{
-    return tr_torrentInfo(cp->tor)->totalSize - cp->sizeNow;
-}
-
 static inline bool tr_cpHasAll( const tr_completion * cp )
 {
